src/locutus/model/terminology.py

```diff
@@ -436,18 +479,51 @@ def add_provenance(
         self, change_type, editor, target=None, timestamp=None, **kwargs
     ):
         if target is None:
-            target = "self"
+            code_index = "self"
+            normalized_target = "self"
+        else:
+            # Ensure special characters in paired targets are handled properly
+            if "|" in target:
+                left, right = target.split("|", 1)
+                # Normalize each side of a paired target
+                normalized_left = (
+                    normalize_ftd_placeholders(left)
+                    if left in FTD_PLACEHOLDERS
+                    else left
+                )
+                normalized_right = (
+                    normalize_ftd_placeholders(right)
+                    if right in FTD_PLACEHOLDERS
+                    else right
+                )
+                normalized_target = generate_paired_string(
+                    normalized_left, normalized_right
+                )
+                # Ensure indexes are formatted properly
+                index_left = get_code_index(left) if left in FTD_PLACEHOLDERS else left

```

Comment:
I feel like this if/else could be migrated into the get_code_index to clean up the calls. By reducing repeated code like this, it makes life vastly easier if you ever need to change something with that logic (ie. only one place to correct instead of several different spots)